File: client/src/app/modules/accounting/services/accounting.journal.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { AccountingJournal } from '../../accounting/model/AccountingJournal';

@Injectable({
  providedIn: 'root'
})
export class AccountingJournalService extends AppRestService<AccountingJournal>{

  constructor(http: HttpClient) {
    super(http, "accounting");
  }

  getAccountingJournals() {
    return this.getListCached(new HttpParams(), "accounting-journals");
  }

  addOrUpdateAccountingJournal(accountingJournal: AccountingJournal) {
    this.clearListCache(new HttpParams(), "accounting-journals");
    return this.addOrUpdate(new HttpParams(), "accounting-journal", accountingJournal, "Enregistré", "Erreur lors de l'enregistrement");
  }

}

File: src/main/java/com/jss/osiris/modules/osiris/accounting/service/AccountingJournalService.java
package com.jss.osiris.modules.osiris.accounting.service;

import java.util.List;

import com.jss.osiris.modules.osiris.accounting.model.AccountingJournal;

public interface AccountingJournalService {
    public List<AccountingJournal> getAccountingJournals();

    public AccountingJournal getAccountingJournal(Integer id);

    public AccountingJournal addOrUpdateAccountingJournal(AccountingJournal accountingJournal);

    public AccountingJournal getAccountingJournalByCode(String code);

}
